lib/fs.js

```diff
@@ -2297,13 +2296,22 @@ function watch(filename, options, listener) {
 
   if (options.persistent === undefined) options.persistent = true;
   if (options.recursive === undefined) options.recursive = false;
-  if (options.recursive && !(isOSX || isWindows))
-    throw new ERR_FEATURE_UNAVAILABLE_ON_PLATFORM('watch recursively');
-  const watcher = new watchers.FSWatcher();
-  watcher[watchers.kFSWatchStart](filename,
-                                  options.persistent,
-                                  options.recursive,
-                                  options.encoding);
+
+  let watcher;
+
+  // TODO(anonrig): Remove this when/if libuv supports it.
```

Comment:
Nit: might be good to be a bit more explicit about removing *what*. Just one of the if branches? Both branches? Not blocking tho.

Reply:
Thanks for the review @jasnell. I'll follow up with a different pull request for this.

Reply:
Followed-up on PR: https://github.com/nodejs/node/pull/45265